src/Files.App/Actions/Open/OpenRepoInVSCodeAction.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) 2023 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+using Files.App.Utils.Shell;
+
+namespace Files.App.Actions
+{
+	internal sealed class OpenRepoInVSCodeAction : ObservableObject, IAction
+	{
+		private readonly IContentPageContext _context;
+
+		private readonly bool _isVSCodeInstalled;
+
+		public string Label
+			=> "OpenRepoInVSCode".GetLocalizedResource();
+
+		public string Description
+			=> "OpenRepoInVSCodeDescription".GetLocalizedResource();
+
+		public bool IsExecutable =>
+			_isVSCodeInstalled &&
+			_context.Folder is not null;
```

Comment:
I think we should disable this action when not in a git repo. At the moment a user can execute it in any folder from the command palette. In this case, the action may not work as expected

Reply:
This should be addressed in [55f41de](https://github.com/files-community/Files/pull/13498/commits/55f41dee3c1059cf2d29796685f9b550bdc5a948)

Reply:
What's the best way to check if the folder is a git repo?

Reply:
There should be a variable in the context

Reply:
Can you confirm it's working for both the root and sub folders?

Reply:
Never mind, we only need one option in the root folder.

Reply:
Hmm, seems to be working now.